### ISSUE TITLE
Unabe to launch in Linux Mint 18.1 Cinnamon 64-bit

### ISSUE THREAD

**Issue by Erkhyan** [olivierkes/manuskript#115]
Ever since I upgraded from 0.3.0 to 0.4.0, I've been unable to run Manuskript.

Running by double-clicking or from a shortcut gives nothing. Running from the Terminal gives this:

    ~/Apps/Manuskript $ ./manuskript
    Traceback (most recent call last):
      File "manuskript", line 11, in <module>
      File "/usr/local/lib/python3.5/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 389, in load_module
      File "manuskript/main.py", line 8, in <module>
      File "/usr/local/lib/python3.5/dist-packages/PyInstaller/loader/pyimod03_importers.py", line 573, in load_module
    ImportError: /usr/lib/x86_64-linux-gnu/mesa/libGL.so.1: undefined symbol: drmGetDevices2
    Failed to execute script manuskript

All the listed required dependencies are installed and up to date.

**Comment by gedakc** [comment]
Hi @Erkhyan, from the provided output I assume that you are running Manuskript for Linux.

Q1)  Which GNU/Linux distribution are you using?
(E.g., Debian, Fedora, Ubuntu, openSUSE, etc.)

Q2) Which method did you use to perform the upgrade?

Method 1)  Using the manuskript-0.4.0-linux.zip file available on the [Manuskript Download](http://www.theologeek.ch/manuskript/download/) page?

Method 2)  Using the instructions to [Run Manuskript from Source Code on Linux](https://github.com/olivierkes/manuskript/wiki/Run-Manuskript-from-Source-Code-on-Linux)?

If using **Method 1** then all of the dependencies _should_ be included in the zip file.  In this case the reference from `./manuskript` to files in `/usr/local/...` seems suspect.

If using **Method 2** then be sure to use the `bin/manuskript` file to invoke manuskript.

**Comment by Erkhyan** [comment]
I'm using Linux Mint 18.1 Cinnamon 64-bit, which is based on Ubuntu 16.04.

I downloaded the `manuskript-0.4.0-linux.zip` file from the download page, then unzipped its contents into a previously emptied folder.

**Comment by gedakc** [comment]
I'm running kubuntu 16.04 and if I `unzip manuskript-0.4.0-linux.zip` into a new folder, I am able to change into the manuskript folder and run `./manuskript`.

Would you be able to try unzipping the *.zip into a brand new folder to see if that works?

**Comment by Erkhyan** [comment]
Brand new folder doesn’t seem to help. I'm getting the same exact error message.

By the way, I just tested running from source (download the `0.4.0.zip` file then run `bin/manuskript`), and it works.

**Comment by gedakc** [comment]
'Glad to hear that you are able to run manuskript 0.4.0 from source code.

To try another test, I unzipped the manuskript-0.4.0-linux.zip file into a new folder on a Virtual Machine running Ubuntu 16.04.  When I change into the manuskript folder and run `./manuskript` I see the following:

```
user@ubuntu1604:~/manuskript$ ./manuskript 
bash: ./manuskript: cannot execute binary file: Exec format error
user@ubuntu1604:~/manuskript$
```

From your experience and this recent test it does appear that there are some issues with the `manuskript-0.4.0-linux.zip` installer.

@olivierkes would you be able to test the `manuskript-0.4.0-linux.zip` installer to see if it works correctly for you?

**Comment by martin3000** [comment]
Same here: trying to run "manuskript" from 0.4.0 package, I get the error 

> ImportError: /usr/lib/x86_64-linux-gnu/mesa/libGL.so.1: undefined symbol: drmGetDevices2
I have Ubuntu 17.04 with intel graphics.

**Comment by gedakc** [comment]
@martin3000, the file `manuskript-0.4.0-linux.zip` is a frozen binary package created with PyInstaller.  It seems that many issues arise when using a PyInstaller frozen binary package on multiple GNU/Linux distributions.  It works on some, requires tweaks on some, and works not at all on others.

Until we have a better way to package _Manuskript_ the best method is to run from source code:  Either from a released source zip file (e.g., `manuskript-0.4.0.zip`), Or from the git repository source code (e.g., **master** or **develop** branch).

Can you try instead to run from source code?

Instructions can be found at [Run Manuskript from Source Code on Linux](https://github.com/olivierkes/manuskript/wiki/Run-Manuskript-from-Source-Code-on-Linux).

**Comment by gedakc** [comment]
@martin3000, alternatively would you be able to try the most recent binary package I created named [manuskript-0.4.0-1-deb9-linux64.zip](https://sourceforge.net/projects/manuskript/files/manuskript/manuskript-0.4.0/manuskript-0.4.0-1-deb9-linux64.zip/download)?

**Comment by martin3000** [comment]
I tried your new zip and it's the same error.

**Comment by gedakc** [comment]
@martin3000, would you be able to try renaming `manuskript/libdrm.so.2` to `manuskript/libdrm.so.2.renamed` and then running manuskript?

For example:
```
cd manuskript
mv libdrm.so.2 libdrm.so.2.renamed
```

The idea came from the this [fman bug report 119](https://github.com/fman-users/fman/issues/119).

**Comment by martin3000** [comment]
I deleted the following files and then it worked. I expected an installer to run but manuskript itself was running, not an installer.

libXau.so.6
libXcursor.so.1
libXinerama.so.1
libxcb-icccm.so.4
libmtdev.so.1
libxcb-render-util.so.0
libXdamage.so.1
libSM.so.6
libxcb-keysyms.so.1
libXext.so.6
libjbig.so.0
libxcb-image.so.0
libxshmfence.so.1
libXxf86vm.so.1
liblz4.so.1
libgudev-1.0.so.0
libmpdec.so.2
libbsd.so.0
libkeyutils.so.1
libidn.so.11
libfontconfig.so.1
libffi.so.6
libwacom.so.2
libatk-1.0.so.0
libpixman-1.so.0
libxcb-xkb.so.1
libxcb-xinerama.so.0
libxcb-xfixes.so.0
libxcb-sync.so.1
libxcb-shm.so.0
libxcb-shape.so.0
libxcb-render.so.0
libxcb-randr.so.0
libxcb-present.so.0
libxcb-glx.so.0
libnettle.so.6
libhogweed.so.4
liblzma.so.5
libdatrie.so.1
libjpeg.so.62
liborc-0.4.so.0
libdouble-conversion.so.1
libQt5Svg.so.5
libselinux.so.1
libdrm.so.2
libtinfo.so.5
libkrb5support.so.0
libkrb5.so.3
libk5crypto.so.3
libgssapi_krb5.so.2
libXfixes.so.3
libXrender.so.1
libXrandr.so.2
libwayland-server.so.0
libwayland-cursor.so.0
libwayland-client.so.0
libcairo.so.2
libcairo-gobject.so.2
libthai.so.0
libQt5WebKitWidgets.so.5
libQt5WebKit.so.5
libgmp.so.10
libp11-kit.so.0
libQt5Quick.so.5
libQt5Qml.so.5
libwebp.so.6
libwebpmux.so.2
libwebpdemux.so.2
libevdev.so.2
libpng16.so.16
libQt5XcbQpa.so.5
libQt5Widgets.so.5
libQt5Sql.so.5
libQt5PrintSupport.so.5
libQt5OpenGL.so.5
libQt5Network.so.5
libQt5Gui.so.5
libQt5EglDeviceIntegration.so.5
libQt5DBus.so.5
libQt5Core.so.5
libgpg-error.so.0
libxkbcommon-x11.so.0
libxkbcommon.so.0
readline.so
mmap.so
libXcomposite.so.1
libepoxy.so.0
libavahi-common.so.3
libavahi-client.so.3
libXi.so.6
libharfbuzz.so.0
libX11-xcb.so.1
libX11.so.6
libz.so.1
libbz2.so.1.0
libcups.so.2
libcom_err.so.2
libproxy.so.1
libgdk_pixbuf-2.0.so.0
libgstreamer-1.0.so.0
libgstbase-1.0.so.0
libgstvideo-1.0.so.0
libgsttag-1.0.so.0
libgstpbutils-1.0.so.0
libgstaudio-1.0.so.0
libgstapp-1.0.so.0
libICE.so.6
libXdmcp.so.6
libinput.so.10
libgthread-2.0.so.0
libgobject-2.0.so.0
libgmodule-2.0.so.0
libglib-2.0.so.0
libgio-2.0.so.0
libpcre.so.3
libpcre16.so.3
libuuid.so.1
libmount.so.1
libblkid.so.1
libgtk-3.so.0
libgdk-3.so.0
libxslt.so.1
libexslt.so.0
libdbus-1.so.3
libpangoft2-1.0.so.0
libpangocairo-1.0.so.0
libpango-1.0.so.0
libicuuc.so.57
libicui18n.so.57
libicudata.so.57
libwayland-egl.so.1
libglapi.so.0
libgbm.so.1
libEGL.so.1
libfreetype.so.6
libgraphite2.so.3
libatk-bridge-2.0.so.0
libatspi.so.0
libreadline.so.7
libstdc++.so.6
libgcc_s.so.1
libtasn1.so.6
libssl.so.1.1
libcrypto.so.1.1
libsqlite3.so.0
libexpat.so.1
libtiff.so.5
libudev.so.1
libsystemd.so.0
libgnutls.so.30
libxml2.so.2
libgcrypt.so.20

**Comment by gedakc** [comment]
Wow, that's a lot of files deleted.  I suspect that at this point manuskript is effectively running from source code using the distro packages for python3, pyqt5, etc.

Out of curiousity, what do the following two commands return as output?

```
uname -a

python3 --version
```

**Comment by martin3000** [comment]
uname gives Linux L1 4.10.0-33-generic #37-Ubuntu SMP Fri Aug 11 10:55:28 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

Python version is 3.5.3

**Comment by gedakc** [comment]
Interesting.  If I understand correctly, Linux Mint 18.1 is based on Ubuntu 16.04.

My Kubuntu 16.04 shows:

```
$ uname -a
Linux kaby 4.4.0-93-generic #116-Ubuntu SMP Fri Aug 11 21:17:51 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
$ python3 --version
Python 3.5.2
```

So it appears that your Linux Mint 18.1 has more recent versions than my Kubuntu 16.04.

Have you only tried the binary packages of Manuskript?
Or have you also tried to [Run Manuskript from Source Code on Linux](https://github.com/olivierkes/manuskript/wiki/Run-Manuskript-from-Source-Code-on-Linux)?

**Comment by martin3000** [comment]
First I did not try "run from source code" because I thought I had to compile a bunch of .c Modules.
Then I downloaded https://github.com/olivierkes/manuskript/archive/master.zip and realized that there is nothing to compile. But it did not run because I don't have PyQt5 and I did not want to install it.

**Comment by gedakc** [comment]
Thank you for confirming that you are only running from the binary files, and for correcting me on the version of Linux you are using (Ubuntu 17.04).

There must be some incompatibility between the packaged libraries (built on Debian 8 and Debian 9) and your Ubuntu 17.04 installation.

It is good to learn that you got manuskript running by removing the above listed libraries.  This may help others in the same situation.

**Comment by martin3000** [comment]
Using the <locate> command I found that many libraries from your package were already installed in my system so I thought this may cause incompatibilities.
How about making a .deb file?

**Comment by martin3000** [comment]
I tried to run manuskript "from source code", I have installed python3-pyqt5.qtwebkit, but I get the error:

> ImportError: No module named 'PyQt5.QtWebKitWidgets'

**Comment by gedakc** [comment]
> How about making a .deb file?

I have investigated the .deb option but have hit a roadblock with PyQt5.  See my **Work-In-Progress** wiki article [Package Manuskript for Linux with StdDeb](https://github.com/olivierkes/manuskript/wiki/Package-Manuskript-for-Linux-with-StdDeb).  I'm all ears on how to create a .deb package.

In an effort to better understand what is going on with Ubuntu 17.04 and Manuskript, I created a 64-bit Virtual Machine with Ubuntu 17.04.  When I unzipped and ran `manuskript-0.4.0-1-linux64.zip` it caused a segmentation fault.

However when I unzipped and ran `manuskript-0.4.0-1-deb9-linux64.zip` it worked!

I haven't yet looked at running a source code install on Ubuntu 17.04.

I wonder what is unique with your Ubuntu 17.04 64-bit install, that `manuskript-0.4.0-1-deb9-linux64.zip` did not work for you?

**Comment by gedakc** [comment]
> I tried to run manuskript "from source code", I have installed python3-pyqt5.qtwebkit, but I get the error:
>>    ImportError: No module named 'PyQt5.QtWebKitWidgets'

I just finished following the steps to [Run Manuskript from Source Code on Linux]() on my Ubuntu 17.04 64-bit VM and ran `bin/manuskript` and manuskript works.

Do you have any `pip3` installed Python3 packages, or are all packages installed via `apt install...`?

**Comment by gedakc** [comment]
> Using the command I found that many libraries from your package were already installed in my system so I thought this may cause incompatibilities. 

This is to be expected.  The binary packages created with PyInstaller intentionally contain a copy of the libraries used.  The intention is that the user need not install Python3, PyQt5, or any other libraries in order to use the binary package.  The binary package is self-contained and is supposed to include all needed libraries, with the exception of `libc` in which the Linux distro version is used.

**Comment by martin3000** [comment]
I have python3-pyqt5 pre-installed by apt. But this version causes a segmentation fault so I did sudo pip3 install PyQt5 and the segmentation fault was fixed. I have a mixture of apt-installed and pip-installed python extensions. When you search for the error `No module named 'PyQt5.QtWebKitWidgets'` you see that many people had this.

**Comment by gedakc** [comment]
I've read that having a combination of apt-installed and pip-installed python extensions can create some issues.

If  you wish to try to avoid `PyQt5.QtWebKitWidgets` altogether we have a **feature/QtWebEngine** branch in git that will try both QtWebKitWidgets and QtWebEngine and use whichever one it finds.  If you want more information on this branch see [PR 116 - Use QtWebEngine instead of deprecated QtWebView](https://github.com/olivierkes/manuskript/pull/116).

The command to switch to the branch is:  `git checkout feature/QtWebEngine`

**Comment by BenMcNevis** [comment]
I had a few issues too. Mint 18.2 KDE. I can start with Manuskript launcher, then added a launcher 'when not running' to panel.

**Comment by gedakc** [comment]
Thanks @BenMcNevis for indicating that you are able to start Manuskript on Mint 18.2 KDE.
 It might help others if you indicate the file name of the manuskript install file you used:

PyInstaller Packages
- [manuskript-0.4.0-linux.zip](https://github.com/olivierkes/manuskript/releases/download/0.4.0/manuskript-0.4.0-linux.zip) - no spell check
- [manuskript-0.4.0-1-linux64.zip](https://sourceforge.net/projects/manuskript/files/manuskript/manuskript-0.4.0/manuskript-0.4.0-1-linux64.zip/download) - includes spell check
- [manuskript-0.4.0-1-deb9-linux64.zip](https://sourceforge.net/projects/manuskript/files/manuskript/manuskript-0.4.0/manuskript-0.4.0-1-deb9-linux64.zip/download) - includes spell check

Source code zip file
- [manuskript-0.4.0.zip](https://github.com/olivierkes/manuskript/archive/0.4.0.zip)
- [git source code](https://github.com/olivierkes/manuskript/wiki/Run-Manuskript-from-Source-Code-on-Linux)

**Comment by gedakc** [comment]
Starting with the 0.6.0 release of Manuskript we now create **.deb**, and **.rpm** files for installation on Debian and Fedora based GNU/Linux distributions respectively.

As such the issues with PyInstaller on GNU/Linux no longer apply.  Hence I am closing this issue.

If issues are encountered with the new **.deb** and **.rpm** packages then please create a new issue and provide details about the problem.

**Comment by gedakc** [comment]
Marking this as **invalid** to prevent it from being automatically included in the CHANGELOG for the upcoming 0.7.0 release.
